File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/C++/s967693306_commented.cpp
#include <iostream>
#include <string>
#include <map>
using namespace std;

void solve() {
    int n, m;
    cin >> n >> m;
    map<int, int> condition;
    for (int i = 0; i < m; ++i) {
        int s, c;
        cin >> s >> c;
        if (condition.find(s) != condition.end()) {
            if (condition[s] != c) {
                cout << -1 << endl;
                return;
            }
        }
        condition[s] = c;
    }

    for (int i = 0; i < 1000; ++i) {
        string s = to_string(i);
        if (s.length() != n) continue;
        bool flag = true;
        for (int j = 0; j < n; ++j) {
            int pos = j + 1;
            int digit = s[j] - '0';
            if (condition.count(pos)) {
                if (digit != condition[pos]) {
                    flag = false;
                    break;
                }
            } else {
                if (j == 0) {
                    if (digit != 1) {
                        flag = false;
                        break;
                    }
                } else {
                    if (digit != 0) {
                        flag = false;
                        break;
                    }
                }
            }
        }
        if (flag) {
            cout << s << endl;
            return;
        }
    }
    cout << -1 << endl;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    solve();
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/Python/s967693306_commented.py
def solve():
    n = int(input())
    m = int(input())
    required = {}
    for _ in range(m):
        s, c = map(int, input().split())
        if s in required:
            if required[s] != c:
                print(-1)
                exit()
        required[s] = c
    for num in range(1000):
        s = str(num)
        if len(s) != n:
            continue
        valid = True
        for j in range(n):
            pos = j + 1
            digit = int(s[j])
            if pos in required:
                if digit != required[pos]:
                    valid = False
                    break
            else:
                if digit != 0:
                    if not (j == 0 and digit == 1):
                        valid = False
                        break
        if valid:
            print(s)
            exit()
    print(-1)

solve()  # End of Code
